be/src/fs/fs_starlet.cpp

```diff
@@ -119,40 +171,85 @@ class StarletFileSystem : public FileSystem {
 
     Type type() const override { return STARLET; }
 
+    StatusOr<Configuration> get_shard_config(int64_t tablet_id) {
+        Configuration conf;
+        ASSIGN_OR_RETURN(auto shardinfo, g_worker->get_shard_info(tablet_id));
+
+        auto iter = shardinfo.properties.find("scheme");
+        if (iter == shardinfo.properties.end()) {
+            // default use starlet s3 filesystem
+            conf.emplace(std::make_pair("scheme", "s3://"));
+        } else {
+            conf.emplace(std::make_pair("scheme", std::move(iter->second)));
+        }
+
+        iter = shardinfo.properties.find(kS3AccessKeyId);
+        if (iter == shardinfo.properties.end()) {
+            return Status::InvalidArgument(
+                    fmt::format("shardinfo {} should contains {} conf", tablet_id, kS3AccessKeyId));
+        }
+        conf.emplace(std::make_pair(kS3AccessKeyId, std::move(iter->second)));
+
+        iter = shardinfo.properties.find(kS3AccessKeySecret);
+        if (iter == shardinfo.properties.end()) {
+            return Status::InvalidArgument(
+                    fmt::format("shardinfo {} should contains {} conf", tablet_id, kS3AccessKeySecret));
+        }
+        conf.emplace(std::make_pair(kS3AccessKeySecret, std::move(iter->second)));
+
+        iter = shardinfo.properties.find(kS3OverrideEndpoint);
+        if (iter == shardinfo.properties.end()) {
+            return Status::InvalidArgument(
+                    fmt::format("shardinfo {} should contains {} conf", tablet_id, kS3OverrideEndpoint));
+        }
+        conf.emplace(std::make_pair(kS3OverrideEndpoint, std::move(iter->second)));
+
+        iter = shardinfo.properties.find(kS3Bucket);
+        if (iter == shardinfo.properties.end()) {
+            return Status::InvalidArgument(fmt::format("shardinfo {} should contains {} conf", tablet_id, kS3Bucket));
+        }
+        conf.emplace(std::make_pair(kS3Bucket, std::move(iter->second)));
+
+        return conf;
+    }
+
     StatusOr<std::unique_ptr<RandomAccessFile>> new_random_access_file(const std::string& path) override {
         return new_random_access_file(RandomAccessFileOptions(), path);
     }
 
     StatusOr<std::unique_ptr<RandomAccessFile>> new_random_access_file(const RandomAccessFileOptions& opts,
```

Comment:
`opts` is not used.
`RandomAccessFileOptions` is replaced with `ReadOptions`?

Reply:
This new_random_access_file is defined by parent class FileSystem